BinaryEncoding.md

```diff
@@ -51,19 +51,26 @@ A single-byte unsigned integer indicating a [value type](AstSemantics.md#types).
 
 # Definitions
 
-### Pre-order encoding
-Refers to an approach for encoding syntax trees, where each node begins with an identifying binary
-sequence, then followed recursively by any child nodes. 
-
-* Examples
-  * Given a simple AST node: `I32Add(left: AstNode, right: AstNode)`
-    * First write the opcode for `I32Add` (uint8)
-    * Then recursively write the left and right nodes.
-
-  * Given a call AST node: `Call(callee_index: uint32_t, args: AstNode[])`
-    * First write the opcode of `Call` (uint8)
-    * Then write the (variable-length) integer `callee_index` (varuint32)
-    * Then recursively write each argument node, where arity is determined by looking up `callee_index` in a table of signatures
+### "Post-order" syntax tree encoding
+
+WASM syntax trees are encoded in a variation of post-order traversal. Nodes with fixed arity will be encoded in post-order. 
+Nodes with variable-arity such as but not limited to blocks, loops, and calls will be delimited with opcode-specific start and end markers. Thus, for fixed arity, the binary sequence begins with child subtrees followed by the opcode.
```

Comment:
That is not the plan of record. Only control flow constructs will be bracketed (to ease the life of one pass compilers or interpreters). Other variadic constructs like calls will simply have an arity immediate, see PR #595.